addons/enemybar/actionTracking.lua

```diff
@@ -0,0 +1,299 @@
+res = require('resources')
+vector = require('vectors')
+packets = require('packets')
+
+tracked_actions = {}
+tracked_enmity = {}
+tracked_debuff = {}
+framerate = 75
+clean_actions_delay = framerate
+clean_actions_tick = clean_actions_delay
+
+function looking_at(a, b)
+	local h = a.facing % math.pi
+	local h2 = (math.atan2(a.x-b.x,a.y-b.y) + math.pi/2) % math.pi
+	return math.abs(h-h2) < 0.15
+end
+
+function is_party_member_or_pet(mob)
+	if not mob then return false end
+	if mob.id == windower.ffxi.get_player().id then return true end
+	local is_pet = mob.is_npc and mob.charmed
+
+	local party = windower.ffxi.get_party()
+	for i=0, party.party1_count-1 do
+		local p = party['p'..i]
+		if p and p.mob then
+			if is_pet then
+				if p.mob.pet_index == mob.index then return true end
+			else
+				if p.mob.id == mob.id then return true end
+			end
+		end
+	end
+	for i=0, party.party2_count-1 do
+		local p = party['a1'..i]
+		if p and p.mob then
+			if is_pet then
+				if p.mob.pet_index == mob.index then return true end
+			else
+				if p.mob.id == mob.id then return true end
+			end
+		end
+	end
+	for i=0, party.party3_count-1 do
+		local p = party['a2'..i]
+		if p and p.mob then
+			if is_pet then
+				if p.mob.pet_index == mob.index then return true end
+			else
+				if p.mob.id == id then return true end
+			end
+		end
+	end
+
+    return false
+end
+
+function handle_action_packet(id, data)
+	if 0x028 == id then
+		local ai = windower.packets.parse_action(data)
+
+		track_enmity(ai)
+		track_actions(ai)
+		track_debuffs(ai)
+	elseif 0x029 then
+    	local message_id = data:unpack('H',0x19)
+    	if not message_id then return end
+    	message_id = message_id%32768
+		local param_1 = data:unpack('I',0x0D)
+    	local target_id = data:unpack('I',0x09)
+		if S{204,206}:contains(message_id) then
+			-- wears off message.
+			local target = windower.ffxi.get_mob_by_id(target_id)
```

Comment:
You get the `target` object here, but you only use it for its `.id` field, which you already have (in the `target_id` variable). `windower.ffxi.get_mob_by_id` is not the slowest function in our API, but it certainly is not fast, it should be avoided if not necessary (so much so that I seriously recommend caching its value when it is not hard to implement).